buildkite/bazel-bench/bazel_bench.py

```diff
@@ -86,23 +111,29 @@ def _get_bazel_commits(date, bazel_repo_path):
     ]
     command_output = subprocess.check_output(args, cwd=bazel_repo_path)
     decoded = command_output.decode("utf-8").split("\n")
+    full_list = [line.strip("'") for line in decoded if line]
 
-    return [line.strip("'") for line in decoded]
+    return _evenly_spaced_sample(full_list, max_commits)
 
 
-def _get_platforms(project_name):
+def _get_platforms(project_name, whitelist):
     """Get the platforms on which this project is run on BazelCI.
+    Filter the results with a whitelist & remove duplicates.
 
     Args:
       project_name: a string: the name of the project. e.g. "Bazel".
+      whitelist: a list of string denoting the whitelist of supported platforms.
 
     Returns:
       A list of string: the platforms for this project.
     """
     http_config = bazelci.DOWNSTREAM_PROJECTS_PRODUCTION[project_name]["http_config"]
     configs = bazelci.fetch_configs(http_config, None)
     tasks = configs["tasks"]
-    return list(map(lambda k: bazelci.get_platform_for_task(k, tasks[k]), tasks))
+    ci_platforms_for_project = list(
```

Comment:
What do you think about using a list comprehension instead?

```python
ci_platforms_for_project = [bazelci.get_platform_for_task(k, tasks[k]) for k in tasks]
```

Reply:
Done